src/plugin/plugin.ts

```diff
@@ -844,13 +1129,29 @@ class Purchases {
    * If user cancelled, userCancelled will be true
    * @param {UpgradeInfo} upgradeInfo Android only. Optional UpgradeInfo you wish to upgrade from containing the oldSKU
    * and the optional prorationMode.
+   * @param {GoogleProductChangeInfo} googleProductChangeInfo Android only. Optional GoogleProductChangeInfo you 
+   * @param {boolean} googleIsPersonalizedPrice Android and Google only. Optional boolean indicates personalized pricing on products available for purchase in the EU.
+   * For compliance with EU regulations. User will see "This price has been customize for you" in the purchase dialog when true.
+   * See https://developer.android.com/google/play/billing/integrate#personalized-price for more info.
    */
   public static purchasePackage(
     aPackage: PurchasesPackage,
     callback: ({productIdentifier, customerInfo,}: { productIdentifier: string; customerInfo: CustomerInfo; }) => void,
     errorCallback: ({error, userCancelled,}: { error: PurchasesError; userCancelled: boolean; }) => void,
-    upgradeInfo?: UpgradeInfo | null
+    upgradeInfo?: UpgradeInfo | null,
+    googleProductChangeInfo?: GoogleProductChangeInfo | null,
+    googleIsPersonalizedPrice: boolean = false
   ) {
+    let oldProductIdentifier = null;
+    let prorationMode = null;
+    if (googleProductChangeInfo !== undefined && googleProductChangeInfo !== null) {
+      oldProductIdentifier = googleProductChangeInfo.oldProductIdentifier;
+      prorationMode = googleProductChangeInfo.prorationMode;
+    } else if (upgradeInfo !== undefined && upgradeInfo !== null) {
```

Comment:
Hmm so `upgradeInfo` is only used if `googleProductChangeInfo` is not passed... I think the best way to do this would be to have 2 methods, one using `upgradeInfo` which is deprecated, and one using the new 2 parameters?

Reply:
Don't think we can have two methods with the same name 😬 Ran into this same issue in React Native as well 😔

Reply:
Ohh yeah... I forgot 😞